src/BarrelCalorimeterSciGlass_geo.cpp

```diff
@@ -209,25 +209,55 @@ static Ref_t create_detector(Detector& lcdd, xml_h handle, SensitiveDetector sen
       const double       length              = family_dim_handle.z_length();
       const auto         flare_angle_polar   = family_dim_handle.attr<double>(_Unicode(flare_angle_polar));
       const unsigned int number              = family_dim_handle.number();
-      const auto         flare_angle_at_face = family_dim_handle.attr<double>(_Unicode(flare_angle_at_face));
+
+      // some per-family calculations rely on extreme values of "beta" to make
+      // conservative estimates
+      double beta_max = beta + flare_angle_polar_prev + (2 * number - 1) * flare_angle_polar;
 
       const double z  = length / 2;
       // Face parameters (see doc/sciglass_tower_front_view.svg for definitions)
       const double y1 = family_dim_handle.y1();
       const double y2 = y1 + length * tan(flare_angle_polar);
       double       x1 = family_dim_handle.x1();
+
+      const auto   flare_angle_at_face_s = family_dim_handle.attr<string>(_Unicode(flare_angle_at_face));
+      double flare_angle_at_face;
+      if (flare_angle_at_face_s == "auto") {
```

Comment:
Is `auto` the only option, or can `auto` be more descriptive? E.g. how you use `pyramidal` instead of `auto` for that value.

Reply:
I was asking myself this question. In principle, one could call it `fit`, although that's still vague about all known and unknown subtle details. I can rename to that, or we could always deprecate the `auto` value if some alternative naming naturally emerges.

Reply:
Sounds good to me. I approved in any case, so feel free to merge.

Reply:
This is good enough for now. Let's merge